### DISCUSSION TITLE
Microsoft Announces VBScript Retirement

### DISCUSSION THREAD

**Discussion by Chiwinks** [ClearFoundry/ClearScript#559] [General]
Hello, all.

In October 2023, Microsoft announced the upcoming announcement of VBScript [here](https://learn.microsoft.com/en-us/windows/whats-new/deprecated-features-resources#vbscript).

How will VBScript retirement impact the ability for ClearScript to continue running VBScript?

**Comment by ClearScriptLib** [comment]
Hi @Chiwinks,

ClearScript doesn't distribute VBScript; it instantiates it by its [Component Object Model (COM)](https://learn.microsoft.com/en-us/windows/win32/com/component-object-model--com--portal) identifier. If VBScript isn't installed or its identifier isn't registered for any reason, instantiation of [`VBScriptEngine`](https://clearscript.clearfoundry.net/Reference/html/T_Microsoft_ClearScript_Windows_VBScriptEngine.htm) should fail gracefully.

Thanks!
